src/components/RoomNameInput.js

```diff
@@ -142,13 +101,9 @@ class RoomNameInput extends Component {
                 onChangeText={(roomName) => {
                     const modifiedRoomName = this.modifyRoomName(roomName);
                     this.setState({roomName: modifiedRoomName});
-                    if (!this.props.shouldShowErrorOnChange) {
-                        return;
-                    }
-                    this.validateRoomName(modifiedRoomName);
```

Comment:
Here. Call `this.props.onChangeText(this.state.roomName);` and I also think we don't need the state for this component as well.

`this.setState({roomName: modifiedRoomName});` This seems unnecessary now.

Reply:
We need the state for modified room name.

Reply:
Ok. let's just focus on `componentDidupdate` part for now. we don't need it. Call `this.props.onChangeText(this.state.roomName);` here.